File: library/src/main/java/cn/wandersnail/widget/listener/RejectableItemClickCallback.java
package cn.wandersnail.widget.listener;

import android.view.View;
import android.widget.AdapterView;

/**
 * 可以控制是否执行点击事件回调
 * <p>
 * date: 2019/8/22 16:40
 * author: zengfansheng
 */
public interface RejectableItemClickCallback {
    default void onReject(AdapterView<?> parent, View view, int position, long id) {}
    
    void onAccept(AdapterView<?> parent, View view, int position, long id);
}

File: library/src/main/java/com/snail/widget/listener/RejectableItemClickCallback.kt
package com.snail.widget.listener

import android.view.View
import android.widget.AdapterView

/**
 * 可以控制是否执行点击事件回调
 *
 * date: 2019-06-16 12:15
 * author: zengfansheng
 */
interface RejectableItemClickCallback : AdapterView.OnItemClickListener {
    fun onReject(parent: AdapterView<*>?, view: View, position: Int, id: Long)

    fun onAccept(parent: AdapterView<*>?, view: View, position: Int, id: Long)
}